plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -194,6 +194,8 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
             closer.register(fileScanTaskIterable);
             this.fileScanTaskIterator = fileScanTaskIterable.iterator();
             closer.register(fileScanTaskIterator);
+            // TODO: this is a workaround until the NPE in https://github.com/apache/iceberg/pull/6322 is fixed
+            isFinished();
```

Comment:
we should have a link to Trino issue to make sure these are removed some day

see https://github.com/trinodb/trino/issues/15371 for example of such an issue

Reply:
Good call, I've created https://github.com/trinodb/trino/issues/15372 and referenced this in the code